In the failing jobs, we're getting errors like these:
```
E       AttributeError: module 'numpy' has no attribute 'astype'. Did you mean: 'dtype'?
```

.. which indicate old numpy versions. I see it mentions installing numpy2.5.1 elsewhere in the same logs. Perhaps it might be using different versions when building and when testing.

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -97,6 +97,7 @@ outputs:
         - array-api-strict
         - array-api-strict <2.4  # [py<312]
         - polars
+        - numpy >=2.2
       source_files:
         - .ci
         - setup.cfg

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -92,7 +92,8 @@ outputs:
         - lightgbm
         - treelite
         - shap
-        - catboost
+        # catboost <1.2.8 is ABI-incompatible with numpy 2.x
+        - catboost >=1.2.8
         - array-api-compat
         - array-api-strict
         - array-api-strict <2.4  # [py<312]

